common/static/coffee/src/discussion/views/discussion_thread_view.coffee

```diff
@@ -22,12 +22,6 @@ if Backbone?
       if @mode not in ["tab", "inline"]
         throw new Error("invalid mode: " + @mode)
 
-      # Quick fix to have an actual model when we're receiving new models from
-      # the server.
-      @model.collection.on "reset", (collection) =>
-        id = @model.get("id")
-        @model = collection.get(id) if collection.get(id)
-
```

Comment:
Is removing this code part of the hotfix or is it just code cleanup? I tried leaving that part of the code in and the thread loading still seems to work fine. If it's only code cleanup, it might be better to do it in a separate PR.